packages/react-reconciler/src/ReactFiberCompleteWork.js

```diff
@@ -361,6 +361,10 @@ function completeWork(
         // If we have an alternate, that means this is an update and we need to
         // schedule a side-effect to do the updates.
         const oldProps = current.memoizedProps;
+        if (oldProps === newProps) {
+          break;
```

Comment:
Be careful with those early returns. I think you’re skipping the ref changing.

Reply:
Can it even change if we bail out?

Reply:
I'll push up a fix tho.

Reply:
In theory we can have a different element with the same props object.